### PR TITLE
Documentation on adding diagnostic output.

### PR DESCRIPTION
**Pull request by dabail10**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
This is only a documentation update.
- [X] Developer(s): 
dabail10 (D. Bailey)
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
I will make sure this builds in readthedocs via the Travis check.
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:
This is a starting point for the tips on adding diagnostics. I am open to suggestions. This addresses issue #317.

### CODE REVIEW

File: doc/source/developer_guide/dg_adding_diagnostics.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Icepack only produces ASCII (text) log output for four points (full model with ITD, an initially ice free point, a land point, and a case with a slab ocean). Each of these files contains the state information for that point. Sometimes additional variables are required in this output. The procedure for adding diagnostic variables is outlined here.
+
+#. For non-BGC variables, one should edit **icedrv\_diagnostics.F90**:
+
+   -  If the variable is already defined within the code, then add it to a "use" statement in the subroutine
+      ``runtime_diags``.
+
+   -  If the variable is just a scalar, then follow the example of "aice". Copy the write statement for Qa to
+      a place in the output list where it is most appropriate. The format "900" is appropriate for most scalars.
+      Edit the copied statement to be the variable you want. The following example adds snow-melt (melts).
```
Comment (eclare108213) [inline doc/source/developer_guide/dg_adding_diagnostics.rst:17]: Here is would add what n is, e.g.

> The following example adds snow melt (melts) for each of the n points.

(no hyphen needed)

File: doc/source/developer_guide/dg_adding_diagnostics.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      index available. Here, you can look at the example of nt_Tsfc. 
+
+   -  In some cases, a new format statement might be required if 900 or 901 are not correct.
+
+#. For BGC variables, one should edit **icedrv\_diagnostics\_bgc.F90**:
+
+   -  If the variable is already defined within the code, then add it to a "use" statement in the subroutine
+      ``hbrine_diags`` or ``bgc_diags`` or ``zsal_diags``. The similar procedure for state variables is used here.
+
+   -  Note that the BGC needs to be activited and the particular tracer turned on.
```
Comment (eclare108213) [inline doc/source/developer_guide/dg_adding_diagnostics.rst:54]: spell activated

**Comment by dabail10** [comment]
Right! I also need to add an example when a variable needs to be derived from other variables.

**Comment by eclare108213** [comment]
This issue has popped back up in the forum, so maybe we should raise its priority...

File: doc/source/developer_guide/dg_adding_diagnostics.rst (hunk 1 above)
Comment (dabail10) [reply doc/source/developer_guide/dg_adding_diagnostics.rst:23]: I've added a bit more here. Hopefully this is helpful.

File: doc/source/developer_guide/dg_adding_diagnostics.rst (hunk 2 above)
Comment (dabail10) [reply doc/source/developer_guide/dg_adding_diagnostics.rst]: Fixed.

**Comment by dabail10** [comment]
I've incorporated Elizabeth's changes here and the readthedocs is building fine. I've also added a fix to the documentation as found by @JFLemieux73.

**Review by eclare108213 — APPROVED** [review]
This is okay. We can continue to improve it as new functionality is added, or as users make specific requests.  E.g. conservation diagnostics are not addressed.

**Comment by dabail10** [comment]
I will wait for @JFLemieux73 to have a look. The conservation diagnostics are on my list.

**Comment by JFLemieux73** [comment]
Thanks Dave for the correction in the section Shortwave radiation: CCSM3. It's ok.

**Pull request merged**